automated-testing/shadow-testing/readme.md

```diff
@@ -0,0 +1,38 @@
+# Shadow Testing
+
+Shadow testing is one of the useful approaches to reduce risks when you consider to replace the current environment (V-Current) with candidate environment with new feature (V-Next). This approach is monitoring and capturing differences between two environments then compare and reduces all risks before you introduce a new feature/release.
+
+This approach is also known as "Shadow Deployment" or "Shadowing Traffic" and similarities with "Dark launching"
+
+In our test cases, code coverage is very important however sometimes providing code coverage can be tricky to replicate real-life combinations and possibilities. In this approach, to test V-Next environment we have side by side deployment, we're replicating the same traffic with V-Current environment and directing same traffic to V-Next environment, the only difference is we don't return any response from V-Next environment to users.
+
+![Shadow Testing Overview](images/shadow-testing.png)
+
+ Referencing back to one of the [Principles of Chaos Engineering](https://principlesofchaos.org/), mentions importance of sampling real traffic like below:
+
+> Systems behave differently depending on environment and traffic patterns. Since the behavior of utilization can change at any time, sampling real traffic is the only way to reliably capture the request path. To guarantee both authenticity of the way in which the system is exercised and relevance to the current deployed system, Chaos strongly prefers to experiment directly on production traffic.
+
+With this Shadow Testing approach we're leveraging real customer behavior in V-Next environment with sampling real traffic and mitigating the risks which users may face on production.
+
+We're testing actual behavior of the product and this cause zero impact to production to test new features since traffic is replicated to V-next environment.
```

Comment:
nit: merge this paragraph with the one above it?

Reply:
Thanks @alkofahi, it's fixed.